pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,433 @@
+use drink::{
+	sandbox_api::assets_api::AssetsAPI,
+	session::{error::SessionError, Session},
+};
+use frame_support::assert_ok;
+use pop_api::{
+	primitives::TokenId,
+	v0::fungibles::events::{Approval, Created, Transfer},
+};
+use pop_sandbox::{Balance, DevnetSandbox as Sandbox, ALICE, BOB};
+use scale::Encode;
+use utils::*;
+
+use super::*;
+
+const TOKEN_ID: TokenId = 1;
+const TOKEN_MIN_BALANCE: Balance = 10_000;
+
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+#[drink::test(sandbox = Sandbox)]
+fn new_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	// Token exists after the deployment.
+	assert!(session.sandbox().asset_exists(&TOKEN_ID));
+	let expected = Created {
+		id: TOKEN_ID,
+		creator: account_id_from_slice(contract.as_ref()),
+		admin: account_id_from_slice(contract.as_ref()),
+	}
+	.encode();
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice()); // Successfully emit event.
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn existing_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+
+	// Fails to deploy contract with a non-existing token ID.
+	assert!(
+		deploy_with_existing_constructor(&mut session, BundleProvider::local()?, TOKEN_ID).is_err()
+	);
+	expect_deployment_reverted(&session, PSP22Error::Custom(String::from("Unknown")));
```

Comment:
Would be nicer if we could combine those helper functions so it becomes:
`assert_eq!(deploy(&mut session, contract, "new", &[TOKEN_ID.to_string()]),  PSP22Error::Custom(String::from("Unknown")))`

This means combining `deploy_with_existing_constructor`, `expect_deployment_reverted`, `deploy_with_new_constructor`. In addition, can we add a oneliner at the top of the tests:
```
let contract = BundleProvider::local()?;
```

The reason for the above is to be as clear as possible in this example. People will have to learn about drink when they see this example so where we can make things a bit easier to understand, lets help.